DistroLauncher/console_service.h

```diff
@@ -63,17 +65,25 @@ namespace Win32Utils
             // the consumer side, both stdout and stderr goes to the same file.
             if (state == consoleState())
                 return;
-            auto res2 = _dup2(state.stdErrFileDescriptor, _fileno(stderrStream));
+            fflush(stderrStream);
+            fflush(stdoutStream);
             SetStdHandle(nStderrHandle, state.stdErrHandle);
-            auto res = _dup2(state.stdOutFileDescriptor, _fileno(stdoutStream));
             SetStdHandle(nStdoutHandle, state.stdOutHandle);
+            auto res2 = _dup2(state.stdErrFileDescriptor, _fileno(stderrStream));
+            auto res = _dup2(state.stdOutFileDescriptor, _fileno(stdoutStream));
         }
 
       public:
         // std::forward<Pipe>(pipe) is just a way to ensure pipe is moved instead of copied.
         ConsoleService(Pipe&& pipe) noexcept : redirectTo{std::forward<Pipe>(pipe)}
         {
             assert(redirectTo.readHandle() != nullptr);
+            // Attempting to find the console window is best-effort if it is not the old style console.
+            // This has been tested for the new Windows Terminal. Other terminals out there may not fit.
+            window_ = FindWindow(L"CASCADIA_HOSTING_WINDOW_CLASS", DistributionInfo::WindowTitle.c_str());
+            if (window_ == NULL) {
+                window_ = GetConsoleWindow();
```

Comment:
I guess this is the function which finds back the default?

Reply:
I tested in a Windows 10 VM launching by double click on the launcher exe and by invoking it from powershell and by the command prompt (both running on the old school console). It falls back nicely. But it has to be the last case, because as I said, GetConsoleWindow() always return something. Checking whether that's not the desired window will be the same amount of work to find the desired window anyway.